tiny_dnn/activations/softmax_layer.h

```diff
@@ -35,8 +35,15 @@ class softmax_layer : public activation_layer {
                            const vec_t &dy) override {
     const serial_size_t len = static_cast<serial_size_t>(dy.size());
 
-    // auxilliary vector to store element wise softmax gradients of all elements
-    vec_t df(len, 0);
+// auxilliary vector to store element wise softmax gradients of all elements
+
+#if HAS_CXX11_THREAD_LOCAL
+#define ATTRIBUTE_TLS thread_local
+#else
+#define ATTRIBUTE_TLS
+#endif
+
+    ATTRIBUTE_TLS vec_t df(len, 0);
```

Comment:
@beru I'm using this snippet for now. We can discuss the possibilities of using `thread_local` at other places - probably in master, instead of this branch too.